File: C++/intTostr.cpp
#include<iostream>
#include<string>
#include<vector>

class Solution{
    public:
    int solve(std::string nums) {
        int n =nums.size();
        if(n==0 || nums[0]=='0'){return 0;}
        std::vector<int>dp(n+1,0);
        dp[0]=1;
        dp[1]=1;
        for(int i=2;i<=n;i++){
            if(nums[i-1] !='0'){
                dp[i] +=dp[i-1];
            }
            if(nums[i-2]=='1'||(nums[i-2]=='2'&&nums[i-1]<='6')){
                dp[i] +=dp[i-2];
            }
        }
        return dp[n];
    }
};

int main(){
    std::string num1="31717126241541717";
    Solution solution;
    std::cout<<solution.solve(num1)<<std::endl;
    return 0;
}

File: python/intTostr.py
class Solution:
    def solve(self , nums: str) -> int:
        n = len(nums)
        if n == 0 or nums[0] == "0":
            return 0
        dp = [0] * (n + 1)
        dp[0] = 1
        dp[1] = 1
        for i in range(2, n+1):
            if nums[i - 1] != "0":
                dp[i] += dp[i - 1]
            if nums[i - 2] == "1" or (nums[i - 2] == "2" and nums[i - 1] <= "6"):
                dp[i] += dp[i - 2]
        return dp[n]

nums="31717126241541717"
s=Solution()
print(s.solve(nums))